File: classes/parte-8-como-funcionan-los-lenguajes/127-la-pila-stack-y-el-marco-de-llamada/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func sumar(n int) int64 {
	if n == 0 {
		return 0
	}
	return int64(n) + sumar(n-1)
}

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	n, _ := strconv.Atoi(strings.TrimSpace(line))
	fmt.Printf("suma=%d profundidad=%d\n", sumar(n), n)
}

File: classes/parte-8-como-funcionan-los-lenguajes/127-la-pila-stack-y-el-marco-de-llamada/implementaciones/c/main.c
#include <stdio.h>

long sumar(long n) {
    return n == 0 ? 0 : n + sumar(n - 1);
}

int main(void) {
    long n;
    if (scanf("%ld", &n) != 1) return 1;
    printf("suma=%ld profundidad=%ld\n", sumar(n), n);
    return 0;
}
